spec/classes/dns_init_spec.rb

```diff
@@ -207,34 +235,68 @@
         ]} }
 
         it do
-          expected = [
-            '// named.conf',
-            "include \"#{rndc_key}\";",
-            'controls  {',
-            '        inet 127.0.0.1 port 953 allow { 127.0.0.1; } keys { "rndc-key"; };',
-            '};',
-            'options  {',
-            "        include \"#{options_path}\";",
-            '};',
-            localzonepath ? "include \"#{localzonepath}\";" : nil,
-            '// additional directives',
-            'logging {',
-            '  channel string {',
-            '    print-severity boolean;',
-            '    print-category boolean;',
-            '  };',
-            '};',
-            'lwres {',
-            '  listen-on [ port integer ] {',
-            '    ( ipv4_address | ipv6_address ) [ port integer ];',
-            '  };',
-            '  view string optional_class;',
-            '  search { string; ... };',
-            '  ndots integer;',
-            '};',
-            '// Public view read by Server Admin',
-            "include \"#{etc_named_directory}/zones.conf\";",
-          ].compact
+
+          if defaultzonepath == 'unmanaged'
+            expected = [
+              '// named.conf',
+              "include \"#{rndc_key}\";",
+              'controls  {',
+              '        inet 127.0.0.1 port 953 allow { 127.0.0.1; } keys { "rndc-key"; };',
+              '};',
+              'options  {',
+              "        include \"#{options_path}\";",
+              '};',
+              localzonepath ? "include \"#{localzonepath}\";" : nil,
+              '// additional directives',
+              'logging {',
+              '  channel string {',
+              '    print-severity boolean;',
+              '    print-category boolean;',
+              '  };',
+              '};',
+              'lwres {',
+              '  listen-on [ port integer ] {',
+              '    ( ipv4_address | ipv6_address ) [ port integer ];',
+              '  };',
+              '  view string optional_class;',
+              '  search { string; ... };',
+              '  ndots integer;',
+              '};',
+              '// Public view read by Server Admin',
+              "include \"#{etc_named_directory}/zones.conf\";",
+            ].compact
+          else
+            expected = [
+              '// named.conf',
+              "include \"#{rndc_key}\";",
+              'controls  {',
+              '        inet 127.0.0.1 port 953 allow { 127.0.0.1; } keys { "rndc-key"; };',
+              '};',
+              'options  {',
+              "        include \"#{options_path}\";",
+              '};',
+              localzonepath ? "include \"#{localzonepath}\";" : nil,
+              defaultzonepath ? "include \"#{defaultzonepath}\";" : nil,
```

Comment:
Rather than duplicating the whole block you could also change this to
```suggestion
              (defaultzonepath && defaultzonepath != 'unmanaged') ? "include \"#{defaultzonepath}\";" : nil,
```

Or you can drop the `unmanaged` bit.

Reply:
Switch to nil. No longer required